### PR TITLE
Translation canary

### PR DESCRIPTION
**Pull request by dashea**
The "translated" tests are not used since the test target is not implemented.

### CODE REVIEW

**Comment by jkonecny12** [comment]
For the first comment apply the same as here https://github.com/rhinstaller/anaconda/pull/620#issuecomment-217808006 .

File: translation-canary/README.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  This contains checks on the strings returned from the translators. This
+  ensures that the content of the translated strings won't break anything.
+  These tests are run on the source directory before creating a release.
+
+Both translatable and translated are run by running the module
+(e.g., `python3 -m translation_canary.translatable`) with the input file(s) as
+the argument.
+
+In addition to the python modules, this project contains xgettext_werror.sh, a
+wrapper for xgettext that treats warnings as errors.  xgettext will print
```
Comment (jkonecny12) [inline translation-canary/README.rst]: I don't know if this is intentional but you have two spaces before *xgettext will print...* .

File: translation-canary/tests/project-tests/test_projects.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public # License and may only be used or replicated with the express permission of # Red Hat, Inc.
```
Comment (jkonecny12) [inline translation-canary/tests/project-tests/test_projects.sh]: Missing newline here.

File: translation-canary/tests/unittests/test_translated.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public # License and may only be used or replicated with the express permission of # Red Hat, Inc.
```
Comment (jkonecny12) [inline translation-canary/tests/unittests/test_translated.py]: Another missing newline char.

**Comment by dashea** [comment]
Updated to fix the spacing issues

**Comment by jkonecny12** [comment]
Looks good to me now.

**Comment by jkonecny12** [comment]
Hi,

I'm closing this PR now. Sorry for missing it for so long time I found this by accident.

Reasons for closing this:

- This is too big copy and paste. I like this functionality but I don't want to add 20 files to enable translations checks.
- It will easily deprecate. Right now there is the projects.txt file with anaconda for Fedora 23 branch. That is too old and it should not be here in the first place. This project is separate from Anaconda and our other projects.
- Weblate can replace at least part of this functionality. Weblate have checks which can be enforced otherwise you can't save the translations. This is for me preferred way because it does not make me to maintain another tests.


Thank you for this idea and I would like to adapt this if it will change to a separate project in the future. And sorry for not solving this sooner.

**Pull request closed without merging**